### PR TITLE
Paraview fix

### PR DESCRIPTION
**Pull request by luigi-calori**
These are the fixes needed  in order to build paraview on my cluster, specifically I installed with
spack install paraview+qt+python+tcl+opengl2%gcc@4.8.2 ^netcdf -mpi

I added support for OpenGL2 backend and avoided to use mesa and mpi as this deployment is aimed at a node with graphics acceleration.
It seems that in order to let ParaView access numpy and matplotlib, these extensions have to be activated:
spack activate py-matplotlib
spack activate py-numpy

### CODE REVIEW

File: var/spack/repos/builtin/packages/netcdf/package.py [hunk 1]
```diff
@@ -9,7 +9,7 @@ class Netcdf(Package):
     homepage = "http://www.unidata.ucar.edu/software/netcdf/"
     url      = "ftp://ftp.unidata.ucar.edu/pub/netcdf/netcdf-4.3.3.tar.gz"
 
-    version('4.4.0', 'f01cb26a0126dd9a6224e76472d25f6c')
+    version('4.4.0', 'cffda0cbd97fdb3a06e9274f7aef438e')
```
Comment (alalazo) [inline var/spack/repos/builtin/packages/netcdf/package.py]: Why did you need to change the md5sum of the package? As far as I can tell the old one is still correct.

Comment (luigi-calori) [inline var/spack/repos/builtin/packages/netcdf/package.py]: While I was building ParaView, I got an error on netcdf deps complaining for a mismatch in MD5 hash: old hash as well as new one were printed, I just copy-pasted the new one in the package and restart the install with success.

I did not check the actual url used, as it is not set explicitely,  spack should compute it.
How we can check? You can build with the old hash?
Is there a better way to check for MD5 ?

Comment (luigi-calori) [inline var/spack/repos/builtin/packages/netcdf/package.py]: I' ve checked now, running:
wget ftp://ftp.unidata.ucar.edu/pub/netcdf/netcdf-4.4.0.tar.gz -O - | md5sum
gives:
cffda0cbd97fdb3a06e9274f7aef438e

Comment (alalazo) [inline var/spack/repos/builtin/packages/netcdf/package.py]: @luigi-calori @tgamblin @adamjstewart : this looks weird to me, unless I am missing something very obvious. I still have in a mirror the `netcdf` downloaded a few days ago:
```
$ wget  ftp://ftp.unidata.ucar.edu/pub/netcdf/netcdf-4.4.0.tar.gz
...
$ ls -l netcdf-4.4.0.tar.gz /home/culpo/mirror/download/netcdf/netcdf-4.4.0.tar.gz
-rw-r--r-- 1 culpo scitas-ge 5078259 29 gen 09:00 /home/culpo/mirror/download/netcdf/netcdf-4.4.0.tar.gz
-rw-r--r-- 1 culpo scitas-ge 5082073 10 feb 19:47 netcdf-4.4.0.tar.gz
$ md5sum netcdf-4.4.0.tar.gz /home/culpo/mirror/download/netcdf/netcdf-4.4.0.tar.gz
cffda0cbd97fdb3a06e9274f7aef438e  netcdf-4.4.0.tar.gz
f01cb26a0126dd9a6224e76472d25f6c  /home/culpo/mirror/download/netcdf/netcdf-4.4.0.tar.gz
```
Unpacking the tarballs in two different folders:
```
$ diff -rq netcdf-4.4.0-new netcdf-4.4.0-old
Files netcdf-4.4.0-new/aclocal.m4 and netcdf-4.4.0-old/aclocal.m4 differ
Files netcdf-4.4.0-new/compile and netcdf-4.4.0-old/compile differ
Files netcdf-4.4.0-new/config.guess and netcdf-4.4.0-old/config.guess differ
Files netcdf-4.4.0-new/config.h.in and netcdf-4.4.0-old/config.h.in differ
Files netcdf-4.4.0-new/config.sub and netcdf-4.4.0-old/config.sub differ
Files netcdf-4.4.0-new/configure and netcdf-4.4.0-old/configure differ
Files netcdf-4.4.0-new/depcomp and netcdf-4.4.0-old/depcomp differ
Files netcdf-4.4.0-new/docs/images/Makefile.in and netcdf-4.4.0-old/docs/images/Makefile.in differ
Files netcdf-4.4.0-new/docs/Makefile.in and netcdf-4.4.0-old/docs/Makefile.in differ
Files netcdf-4.4.0-new/examples/C/Makefile.in and netcdf-4.4.0-old/examples/C/Makefile.in differ
Files netcdf-4.4.0-new/examples/CDL/Makefile.in and netcdf-4.4.0-old/examples/CDL/Makefile.in differ
Files netcdf-4.4.0-new/examples/Makefile.in and netcdf-4.4.0-old/examples/Makefile.in differ
Files netcdf-4.4.0-new/h5_test/Makefile.in and netcdf-4.4.0-old/h5_test/Makefile.in differ
Files netcdf-4.4.0-new/include/Makefile.in and netcdf-4.4.0-old/include/Makefile.in differ
Files netcdf-4.4.0-new/include/netcdf_meta.h and netcdf-4.4.0-old/include/netcdf_meta.h differ
Files netcdf-4.4.0-new/install-sh and netcdf-4.4.0-old/install-sh differ
Files netcdf-4.4.0-new/libdap2/Makefile.in and netcdf-4.4.0-old/libdap2/Makefile.in differ
Files netcdf-4.4.0-new/libdispatch/Makefile.in and netcdf-4.4.0-old/libdispatch/Makefile.in differ
Files netcdf-4.4.0-new/liblib/Makefile.in and netcdf-4.4.0-old/liblib/Makefile.in differ
Files netcdf-4.4.0-new/libsrc/Makefile.in and netcdf-4.4.0-old/libsrc/Makefile.in differ
Files netcdf-4.4.0-new/libsrc4/Makefile.in and netcdf-4.4.0-old/libsrc4/Makefile.in differ
Files netcdf-4.4.0-new/libsrcp/Makefile.in and netcdf-4.4.0-old/libsrcp/Makefile.in differ
Files netcdf-4.4.0-new/ltmain.sh and netcdf-4.4.0-old/ltmain.sh differ
Files netcdf-4.4.0-new/m4/libtool.m4 and netcdf-4.4.0-old/m4/libtool.m4 differ
Files netcdf-4.4.0-new/m4/lt~obsolete.m4 and netcdf-4.4.0-old/m4/lt~obsolete.m4 differ
Files netcdf-4.4.0-new/m4/ltoptions.m4 and netcdf-4.4.0-old/m4/ltoptions.m4 differ
Files netcdf-4.4.0-new/m4/ltsugar.m4 and netcdf-4.4.0-old/m4/ltsugar.m4 differ
Files netcdf-4.4.0-new/m4/ltversion.m4 and netcdf-4.4.0-old/m4/ltversion.m4 differ
Files netcdf-4.4.0-new/Makefile.in and netcdf-4.4.0-old/Makefile.in differ
Files netcdf-4.4.0-new/missing and netcdf-4.4.0-old/missing differ
Files netcdf-4.4.0-new/ncdap_test/expected3/Makefile.in and netcdf-4.4.0-old/ncdap_test/expected3/Makefile.in differ
Files netcdf-4.4.0-new/ncdap_test/expected4/Makefile.in and netcdf-4.4.0-old/ncdap_test/expected4/Makefile.in differ
Files netcdf-4.4.0-new/ncdap_test/expectremote3/Makefile.in and netcdf-4.4.0-old/ncdap_test/expectremote3/Makefile.in differ
Files netcdf-4.4.0-new/ncdap_test/expectremote4/Makefile.in and netcdf-4.4.0-old/ncdap_test/expectremote4/Makefile.in differ
Files netcdf-4.4.0-new/ncdap_test/Makefile.in and netcdf-4.4.0-old/ncdap_test/Makefile.in differ
Files netcdf-4.4.0-new/ncdap_test/testdata3/Makefile.in and netcdf-4.4.0-old/ncdap_test/testdata3/Makefile.in differ
Files netcdf-4.4.0-new/ncdump/cdl/Makefile.in and netcdf-4.4.0-old/ncdump/cdl/Makefile.in differ
Files netcdf-4.4.0-new/ncdump/ctest.c and netcdf-4.4.0-old/ncdump/ctest.c differ
Files netcdf-4.4.0-new/ncdump/expected/Makefile.in and netcdf-4.4.0-old/ncdump/expected/Makefile.in differ
Files netcdf-4.4.0-new/ncdump/Makefile.in and netcdf-4.4.0-old/ncdump/Makefile.in differ
Files netcdf-4.4.0-new/ncgen/Makefile.in and netcdf-4.4.0-old/ncgen/Makefile.in differ
Files netcdf-4.4.0-new/ncgen3/Makefile.in and netcdf-4.4.0-old/ncgen3/Makefile.in differ
Files netcdf-4.4.0-new/nctest/Makefile.in and netcdf-4.4.0-old/nctest/Makefile.in differ
Files netcdf-4.4.0-new/nc_test/Makefile.in and netcdf-4.4.0-old/nc_test/Makefile.in differ
Files netcdf-4.4.0-new/nc_test4/Makefile.in and netcdf-4.4.0-old/nc_test4/Makefile.in differ
Files netcdf-4.4.0-new/oc2/Makefile.in and netcdf-4.4.0-old/oc2/Makefile.in differ
Files netcdf-4.4.0-new/test-driver and netcdf-4.4.0-old/test-driver differ
```
It seems the sources were patched and silently substituted without bumping the patch version...

**Comment by tgamblin** [comment]
What's different about the files?  I'm willing to trust `netcdf.org` and bump to the new version, although it would be nice to know exactly what changed.

**Comment by alalazo** [comment]
@tgamblin I was not thinking at security issues. I just find this a somewhat strange releasing procedure : two different tarballs are **both** the latest stable release 4.4.0...

**Comment by tgamblin** [comment]
@alalazo: I agree but it looks like they already blew the old one away.  Sigh.  I think we have to integrate the new checksum.

**Pull request merged**